test/DeltaTriplesTest.cpp

```diff
@@ -0,0 +1,269 @@
+// Copyright 2023 - 2024, University of Freiburg
+//  Chair of Algorithms and Data Structures.
+//  Authors:
+//    2023 Hannah Bast <bast@cs.uni-freiburg.de>
+//    2024 Julian Mundhahs <mundhahj@tf.uni-freiburg.de>
+
+#include <gtest/gtest.h>
+
+#include "./util/GTestHelpers.h"
+#include "./util/IndexTestHelpers.h"
+#include "absl/strings/str_split.h"
+#include "engine/ExportQueryExecutionTrees.h"
+#include "index/DeltaTriples.h"
+#include "index/IndexImpl.h"
+#include "index/Permutation.h"
+#include "parser/RdfParser.h"
+
+namespace Matchers {
+inline auto InAllPermutations = [](auto InnerMatcher) {
+  return testing::AllOfArray(ad_utility::transform(
+      Permutation::ALL, [&InnerMatcher](const Permutation::Enum& perm) {
+        return testing::ResultOf(
+            absl::StrCat(".getLocatedTriplesPerBlock(",
+                         Permutation::toString(perm), ")"),
+            [perm](const DeltaTriples& deltaTriples) {
+              return deltaTriples.getLocatedTriplesPerBlock(perm);
+            },
+            InnerMatcher);
+      }));
+};
+inline auto NumTriplesInAllPermutations =
+    [](size_t expectedNumTriples) -> testing::Matcher<const DeltaTriples&> {
+  return InAllPermutations(AD_PROPERTY(LocatedTriplesPerBlock,
+                                       LocatedTriplesPerBlock::numTriples,
+                                       testing::Eq(expectedNumTriples)));
+};
+inline auto NumTriples =
+    [](size_t inserted, size_t deleted,
+       size_t inAllPermutations) -> testing::Matcher<const DeltaTriples&> {
+  return testing::AllOf(
+      AD_PROPERTY(DeltaTriples, DeltaTriples::numInserted,
+                  testing::Eq(inserted)),
+      AD_PROPERTY(DeltaTriples, DeltaTriples::numDeleted, testing::Eq(deleted)),
+      NumTriplesInAllPermutations(inAllPermutations));
+};
+}  // namespace Matchers
+namespace m = Matchers;
+
+// Fixture that sets up a test index.
+class DeltaTriplesTest : public ::testing::Test {
+ protected:
+  // The triples in our test index (as a `std::vector` so that we have easy
+  // access to each triple separately in the tests below).
+  static constexpr const char* testTurtle =
+      "<a> <upp> <A> . "
+      "<b> <upp> <B> . "
+      "<c> <upp> <C> . "
+      "<A> <low> <a> . "
+      "<B> <low> <b> . "
+      "<C> <low> <c> . "
+      "<a> <next> <b> . "
+      "<b> <next> <c> . "
+      "<A> <next> <B> . "
+      "<B> <next> <C> . "
+      "<b> <prev> <a> . "
+      "<c> <prev> <b> . "
+      "<B> <prev> <A> . "
+      "<C> <prev> <B>";
+
+  // Query execution context with index for testing, see `IndexTestHelpers.h`.
+  QueryExecutionContext* testQec = ad_utility::testing::getQec(testTurtle);
+
+  // Make `TurtleTriple` from given Turtle input.
+  std::vector<TurtleTriple> makeTurtleTriples(
+      std::vector<std::string> turtles) {
+    RdfStringParser<TurtleParser<Tokenizer>> parser;
+    // turtles is copied here
```

Comment:
And why is that? Can you just take it by `const&` and drop the comment. If the copy is require,d please explain in a comment why that is.